### PR TITLE
Feature/add skybox

### PR DESCRIPTION
**Pull request by carobouette**
add skybox

### CODE REVIEW

File: Projet/render/GameFactory.cpp [hunk 1]
```diff
@@ -77,7 +77,13 @@ void GameFactory::CreatePlayer(Math::Transform Transform)
 	auto PlayerBody = MyPlayer->AddComponent<RigidBody>(RigidBody::RigidActorType::Dynamic);
 	PlayerBody->SetMass(10.f);
 
+	PM3D::CMoteurWindows::GetInstance().SetSkyBox(new Skybox{
+	&MyPlayer->Transform, PM3D::CMoteurWindows::GetInstance().GetResourcesManager().GetMesh(L".\\modeles\\sky\\sky.OMB"),
+	PM3D::CMoteurWindows::GetInstance().GetResourcesManager().GetShader(L".\\shaders\\MiniPhong.fx") });
```
Comment (baptistegasser) [inline Projet/render/GameFactory.cpp]: Why get a shader ?

File: Projet/render/Moteur.h [hunk 2]
```diff
@@ -159,6 +160,7 @@ template <class T, class TDevice> class CMoteur :public Singleton<T>
 	CDIManipulateur& GetGestionnaireDeSaisie() { return GestionnaireDeSaisie; }
 	ResourcesManager& GetResourcesManager() { return ResourcesManager; }
 	Scene& GetScene() { return *CurrentScene; }
+	void SetSkyBox(Pitbull::Actor* Box) { SkyBox = std::unique_ptr<Pitbull::Actor>(std::move(Box)); }
```
Comment (baptistegasser) [inline Projet/render/Moteur.h]: There is no need to move the Box ptr here

File: Projet/render/Moteur.h [hunk 3]
```diff
@@ -303,6 +307,7 @@ template <class T, class TDevice> class CMoteur :public Singleton<T>
 
 	// La seule scène
 	std::shared_ptr<Scene> CurrentScene;
+	std::unique_ptr<Pitbull::Actor> SkyBox;
```
Comment (baptistegasser) [inline Projet/render/Moteur.h]: Should the skybox be stored in the scene rather than the Engine ?

File: Projet/render/GameFactory.cpp (hunk 1 above)
Comment (carobouette) [reply Projet/render/GameFactory.cpp]: to render the texture and the lights, the skybox is a obj

File: Projet/render/Moteur.h (hunk 3 above)
Comment (carobouette) [reply Projet/render/Moteur.h]: i changed

File: Projet/render/Moteur.h (hunk 2 above)
Comment (carobouette) [reply Projet/render/Moteur.h]: look at the addActor in the scene, i did a std::move but i am not sure it is usefull

**Review by baptistegasser — APPROVED** [review]
let's gooo

**Pull request merged**